src/stm32/i2c.c

```diff
@@ -51,18 +51,69 @@ static const struct i2c_info i2c_bus[] = {
 #endif
 };
 
+static void
+i2c_us_delay(uint32_t us)
+{
+    uint32_t t = timer_read_time() + timer_from_us(us);
+    while(t > timer_read_time());
```

Comment:
This is generally not correct because timer could and will overflow.
Please use `timer_is_before()`:
```
    while (timer_is_before(timer_read_time(), end))
        ;
```

Reply:
Oops, yes. I've actually written this function a few times now in various places. Feels like I should just add a `spin_us(usec)` function somewhere?

Reply:
Sometimes, I also wish for it.
But it so happened historically that local pauses are made locally.